frontend/src/components/module-editor/LessonItem.tsx

```diff
@@ -58,7 +101,29 @@ const LessonItem = ({
               fontSize="18px"
               size="sm"
               icon={<EditIcon />}
+              onClick={onOpen}
             />
+
+            <Modal
+              header="Edit lesson title"
+              isOpen={isOpen}
+              onConfirm={() => updateLessonName(title)}
+              onCancel={() => {
+                resetState();
+                onClose();
+              }}
+            >
+              <TextInput
```

Comment:
Would be good to specify the `defaultValue` prop so the field is initialized with the current title. If you do this, it would also be good to _not_ reset the `title` state so that the default state still works on subsequent modal displays.